### PR TITLE
@enhancement/86epe339x/SATA_Questionnaire_blocks fit errorhandiler sa…

### PR DESCRIPTION
**Pull request by acostales08**
…ta validation and reimport

### CODE REVIEW

File: apps/simulator/src/core/utils/useErrorhandler.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { FormHelperText } from '../../components/FormHelperText'; 
+
+type ErrorHandlerProps = {
+    isValid: boolean;
+    errorMessage: string;
+};
+
+export const useErrorHandler = ({ isValid, errorMessage }: ErrorHandlerProps) => {
```
Comment (DropitupQwerty) [inline apps/simulator/src/core/utils/useErrorhandler.tsx:9]: Add return types for code readability

File: apps/simulator/src/components/blocks/RegularQuestions/SATA/RegSATAQuestion.tsx [hunk 2]
```diff
@@ -1,113 +1,102 @@
...
+
+  return (
+    <div className='h-full px-10 py-5'>
+      <Paper elevation={3}>
+        <FormProvider {...form}>
+          {itemselection?.length > 0 && itemselection.map((item: datatypes.CalcItemSelectValues, itemIndex) => (
+            <div key={itemIndex} className="py-8 px-16">
+              <p className="p-2 py-4">{item.question}</p>
+                <div className="">
+                {contents.answerUI?.length > 0 && contents.answerUI.map((answerUImap, answerUIidx) => {
```
Comment (DropitupQwerty) [inline apps/simulator/src/components/blocks/RegularQuestions/SATA/RegSATAQuestion.tsx:64]: Change naming convention for answerUIidx

File: apps/simulator/src/components/blocks/RegularQuestions/SATA/RegSATAQuestion.tsx [hunk 3]
```diff
@@ -1,113 +1,102 @@
...
+                                __html: answerUImap.answerInstruction,
+                              }}
+                            />
+                          </p>
+                      </div>          
+                  )        
+                })}
+                <div className='px-4'>
+                  {fields?.length > 0 &&
+                    fields.map((choices: any, idx: any) => (
```
Comment (DropitupQwerty) [inline apps/simulator/src/components/blocks/RegularQuestions/SATA/RegSATAQuestion.tsx:83]: avoid use of any types

File: apps/simulator/src/components/blocks/RegularQuestions/SATA/RegSATAQuestion.tsx [hunk 4]
```diff
@@ -1,113 +1,102 @@
...
+import { useFormSubmissionBindingHooks } from '@repo/utils/hooks/useFormSubmissionBindingHooks';
+import { useErrorHandler } from '@/core/utils/useErrorhandler';
+
+interface Props extends RegularQuestion {
+    regSataAtom: RegularSATAValidationType | undefined
+    handleSubmit: (value: RegularSATAValidationType) => void
+    ParsedChoices: any
+}
+
+export const RegSATAQuestion: React.FC<Props> = ({ handleSubmit, regSataAtom, itemselection, contents, ParsedChoices }) => {
```
Comment (DropitupQwerty) [inline apps/simulator/src/components/blocks/RegularQuestions/SATA/RegSATAQuestion.tsx:24]: Change this naming convention.
use SATA.tsx for rendering and
and SATABlocks.tsx for logic

**Review by JMSevilla — APPROVED** [review]

**Review by DropitupQwerty — APPROVED** [review]

**Review by JezChristian — APPROVED** [review]

**Pull request merged**